napari/layers/_vectors_layer/model.py

```diff
@@ -89,7 +95,10 @@ def _get_shape(self):
         if len(self.data) == 0:
             return np.ones(self.data.ndim, dtype=int)
         else:
-            return np.max(self.data[:, 0, :], axis=0) + 1
+            data = copy(self.data)
+            data[:, 1, :] = data[:, 0, :] + self.length * data[:, 1, :]
+            maxs = np.max(data, axis=(0, 1)) + 1
+            return maxs
```

Comment:
I'm a little confused about what was happening here (why the +1?) and very confused about what is happening now!

Reply:
have removed the `+1` - wasn't needed, and removed the duplication with the code below.